control-plane/agents/src/bin/core/node/specs.rs

```diff
@@ -183,4 +185,94 @@ impl ResourceSpecsLocked {
         registry.store_obj(&drained_node_spec).await?;
         Ok(drained_node_spec)
     }
+
+    /// Add the draining volume to the node spec for checking shutdown nexuses.
+    pub(crate) async fn add_node_draining_volumes(
+        &self,
+        registry: &Registry,
+        node_id: &NodeId,
+        draining_volumes: &HashSet<VolumeId>,
```

Comment:
I think we can consume by value here?

Reply:
Fixed. Did the same for remove_node_draining_volumes().